sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveUnion.scala

```diff
@@ -68,7 +90,13 @@ object ResolveUnion extends Rule[LogicalPlan] {
     } else {
       left
     }
-    Union(leftChild, rightChild)
+    val newUnion = Union(leftChild, rightChild)
+    if (newUnion.allChildrenCompatible) {
```

Comment:
NOTE: In case of non-compatible inputs, we don't fill `outputUnon` here for keeping the current error handing in `CheckAnalysis`.